### ISSUE TITLE
Do an audit of requirements files, pyproject.toml, ci config, tox

### ISSUE THREAD

**Issue by willingc** [nteract/papermill#747]
We seem to have added some complexity and it probably makes sense to audit these workflows.

**Comment by willingc** [comment]
If I am understanding correctly, everything in `requirements` folder is dev and extras. The `requirements.txt` is the base requirements to build papermill. The `docs/requirements.txt` should have both the docs requirements and the package requirements which doesn't currently work. Does that align with other's understanding.

**Comment by willingc** [comment]
I'm going to when I have time...try to switch this to the more modern packaging format. I think if we use hatch we can get rid of a bunch of stuff including tox and slim down the config files and lean more into pyproject.toml. This should improve maintainability.

**Comment by MSeal** [comment]
Yes I think Python3.13 might force the issue anyway, but I agree we need to redo the setup.py / requirements pattern.

**Comment by Borda** [comment]
Would you be in favor of moving out of tox? and using pure `pytest` and GHA?

**Comment by ofek** [comment]
Maintainer of Hatch and Hatchling here! Let me know if I can be of assistance 🙂

**Comment by willingc** [comment]
Replaced with #833
